widget/tabcontainer_test.go

```diff
@@ -56,3 +57,52 @@ func TestTabContainer_ApplyTheme(t *testing.T) {
 	Renderer(tabs).ApplyTheme()
 	assert.NotEqual(t, barColor, underline.FillColor)
 }
+
+func TestTabContainerRenderer_Layout(t *testing.T) {
+	tabs := NewTabContainer(
+		NewTabItemWithIcon("Text1", theme.CancelIcon(), canvas.NewCircle(theme.BackgroundColor())),
+		NewTabItemWithIcon("Text2", theme.ConfirmIcon(), canvas.NewCircle(theme.BackgroundColor())),
+	)
+
+	r := Renderer(tabs)
+	r.Layout(fyne.NewSize(100, 100))
+
+	images, texts := collectImagesAndTexts(r.Objects(), fyne.NewPos(0, 0))
```

Comment:
I'm not really a fan of this approach. I understand it is working around not wanting to be testing the details of a button renderer within our tabs...
Which makes me wonder if we should be testing at this level... would it not be enough to simply test that the buttons created have an Icon?

Reply:
This, again, implies that we use buttons at all. But in one or two PRs this maybe isn't the case anymore.

Reply:
yes, but for now is when the tests are being added?

Reply:
That doesn't make it any better.

Reply:
I'm not sure I understand. Whilst we do use buttons the tabContainerRenderer does not need to test layouts that are outwith it's control. 
If the future renderer for tabContainer starts laying out icons then we can re-introduce this test.

Reply:
I want to test the behaviour of the tab container regarding positioning of tab contents.
That's exactly what I'm doing here. It doesn't matter if the tab container gives the control to someone else (the button) or not.
I cannot see where it helps to test that we depend on a button to do the work. The button is not part of the requirement.

Reply:
I do understand your point, but I am not sure I agree. This test ensures that the tabs are laid out in a certain way. Is that the layout that we want for tabs, and therefore are testing? My understanding was that buttons were in use "for now" but I realise I could be confusing this with other conversations.

Reply:
Buttons _are_ in use _for now_. But that does not mean that this is a requirement. That's why the test must not assume that there is a button.

But having the possibility to add an icon to a tab which then is displayed left to the text is a requirement. That's why it is tested.

Reply:
That is a very good point. Let's keep in mind that it would be good to refactor this method once different ways of walking etc are in place - or if we see this being used in other tests. I will mark this resolved but the other two comments I think do need the changes we discussed.